openedx/core/djangoapps/catalog/utils.py

```diff
@@ -76,6 +81,34 @@ def get_programs(user=None, uuid=None, type=None):  # pylint: disable=redefined-
         return []
 
 
+def get_programs_for_credentials(user, programs_credentials):
+    """ Given a user and an iterable of credentials, get corresponding programs
+    data and return it as a list of dictionaries.
+
+    Arguments:
+        user (User): The user to authenticate as for requesting programs.
+        programs_credentials (list): List of credentials awarded to the user
+            for completion of a program.
+
+    Returns:
+        list, containing programs dictionaries.
+    """
+    certificate_programs = []
+
+    programs = get_programs(user)
```

Comment:
Here's something I've been thinking about. I wouldn't make changes as part of this PR to address the following, but since you're in here, we should talk about it.

`get_programs` retrieves every program available in the catalog and caches that data. This has worked well enough with a small number of programs, but isn't going to scale well as that number grows. If it gets slow enough, we may end up with cache stampedes like the one we saw when we introduced MicroMasters as many different processes try to make this same series of requests when the cached data expires.

If we were to add an API endpoint to the catalog which returns only those programs associated with the user's enrolled course runs, we could implement a function on this end which compares cached programs against the course runs the user is enrolled in, then only requests those missing from the cache.